### PR TITLE
Add image metadata support

### PR DESCRIPTION
**Pull request by ausi**
Adds a configuration option to specify which metadata fields should be kept in the images, e.g.:

```yaml
contao:
    image:
        preserve_metadata:
            exif: { IFD0: [Copyright, Artist] }
            iptc: ['2#116', '2#080', '2#115', '2#110']
```

And displays the metadata in the file manager:

filemanager-metadata

### CODE REVIEW

**Comment by Toflar** [comment]
Shouldn't that be part of the image sizes config too? I mean, maybe we need it in the FE for use case A but not B?🤔
Of course for the BE there needs to be the current setting as well😊

**Comment by ausi** [comment]
> Shouldn't that be part of the image sizes config too?

It already is, see https://github.com/contao/contao/pull/5837/files#diff-9b94f76b99d33315fb56d2727409c02c2fa453692699d8579a60b650f55746efR319-R327
But I only added it to the image sizes that are configured in the *config.yaml*. 
Do you think this should be available in the `tl_image_size` DCA too?

**Comment by Toflar** [comment]
Yes that's what I meant :)

File: core-bundle/contao/controllers/BackendPopup.php
Comment (Toflar) [inline core-bundle/contao/controllers/BackendPopup.php]: Looks like something that could go to `contao/image`? 🤔

**Comment by ausi** [comment]
> Shouldn't that be part of the image sizes config too?

Done in 470e5da161693d4691f39824104004898a09c6cc

File: core-bundle/contao/controllers/BackendPopup.php
Comment (ausi) [reply core-bundle/contao/controllers/BackendPopup.php]: I’m not sure. Providing proper labels does not have to be part of the library I think. If it would be it should IMO work for all metadata formats (and be more complete) and that is properly way too much work.

I just see this as an UX improvment for your backend users.

Comment (Toflar) [reply core-bundle/contao/controllers/BackendPopup.php]: Yeah which I think is definitely part of the library. It doesn't have to be translated or anything as it is mainly technical. But `City` is always better than `2#090`, right? If I'm using `contao/image` in another project, I would want `City` there too 😇

File: core-bundle/contao/dca/tl_image_size.php
Comment (Toflar) [inline core-bundle/contao/dca/tl_image_size.php:454]: Are we sure that we want to store `serialize(array($key => $value))` here? I don't know of a better alternative atm but just wondering if that's the best we can do?

File: composer.json
Comment (leofeyer) [inline composer.json]: Any chance that we can release a stable version before merging this PR?

File: core-bundle/contao/dca/tl_image_size.php
Comment (Toflar) [reply core-bundle/contao/dca/tl_image_size.php:454]: Also, I'd actually love to see more listeners for the options rather than our funny `tl_*` classes.

File: composer.json
Comment (ausi) [reply composer.json]: Yes we should. But I’d like to have more feedback from the community first, see https://github.com/contao/image/issues/80#issuecomment-1435661044

**Comment by Toflar** [comment]
@ausi I think there's not much more feedback to be expected in `contao/image`. Let's get this ready for 5.2, shall we? ☺️

File: core-bundle/contao/controllers/BackendPopup.php
Comment (ausi) [reply core-bundle/contao/controllers/BackendPopup.php]: Added a `toReadable()` method in https://github.com/contao/image/pull/96

File: core-bundle/contao/dca/tl_image_size.php
Comment (ausi) [reply core-bundle/contao/dca/tl_image_size.php:454]: > I don't know of a better alternative atm but just wondering if that's the best we can do?

I think this should be good enough. Creating a seperate child table/DCA seems overkill to me.

**Comment by ausi** [comment]
~~After contao/image 1.2.0 is released (https://github.com/contao/image/pull/100)~~ this should be RTM I think ~~(after changing the dependency to `"contao/image": "^1.2"`).~~
__Update:__ version 1.2.0 is released: https://github.com/contao/image/releases/tag/1.2.0

File: composer.json
Comment (ausi) [reply composer.json]: Done in e07fe35da1a8c972cff193a990518412c1793527

**Review by Toflar — COMMENTED** [review]
Conflicts need to resolved for a final approval :)

**Review by Toflar — DISMISSED** [review]
As always - a wonderful addition to Contao, thank you so much @ausi ❤️

**Review by leofeyer — COMMENTED** [review]
This looks and works great! 🎉 Just two minor nitpicks:



In German, the captions should be `EXIF-Metadaten` but the way it is currently implemented, it could only be `EXIF -Metadaten`. Can we adjust the code so a correct translation is possible?

What about the empty sections and values `IPTC IIM Metadaten` and `dc:subject`. Does it make sense to show them even though they are empty?

**Comment by ausi** [comment]
> In German, the captions should be `EXIF-Metadaten` but the way it is currently implemented, it could only be `EXIF -Metadaten`. Can we adjust the code so a correct translation is possible?

Done in 4ff32acfdf01a1ee061342ee5604c9deecd66df4

> What about the empty sections and values `IPTC IIM Metadaten` and `dc:subject`. Does it make sense to show them even though they are empty?

Done in ac78bcaa4d7ade08dbffc02db44a5b31c1f63b61

File: core-bundle/contao/languages/en/default.xlf
Comment (leofeyer) [inline core-bundle/contao/languages/en/default.xlf]:
```suggestion
        <source>%s metadata</source>
```

File: core-bundle/contao/templates/backend/be_popup.html5
Comment (leofeyer) [inline core-bundle/contao/templates/backend/be_popup.html5]:
```suggestion
              <?php if (count(array_filter($values))): ?>
                <tr>
                  <td><span class="tl_label"><?= Contao\StringUtil::specialchars($label) ?></span></td>
                  <td><?= Contao\StringUtil::specialchars(implode(', ', $values)) ?></td>
                </tr>
              <?php endif; ?>
``````

This matches the template coding style a bit more. 🙈 I haven‘t tested the `count(array_filter($values))` part though.

Comment (ausi) [reply core-bundle/contao/templates/backend/be_popup.html5]: `count(array_filter($values, \strlen(...)))` should work I think because the string `'0'` should not be filtered out.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**
